src/components/AlertConfirm/AlertConfirm.tsx

```diff
@@ -0,0 +1,75 @@
+import * as React from 'react';
+import { bem } from '../../utils';
+import { Heading } from '../Heading';
+import { Button } from '../Buttons';
+import { Modal, ModalProps } from '../Modal';
+import styles from './AlertConfirm.scss';
+
+type ButtonProps = {
+    onClick: () => void;
+    label: string;
+};
+
+interface DialogProps extends ModalProps {
+    /** Should the alert appear */
+    isOpen: boolean;
+    /** The alert content */
+    children: ReactNode;
+    /** Dialog title */
+    title?: string;
+    /** Properties of the accept button */
+    acceptButton: ButtonProps;
+    /** Properties of the cancel button */
+    cancelButton?: ButtonProps;
+}
+
+export interface AlertProps extends DialogProps {
+    /** Alerts do not have cancel button */
+    cancelButton?: never;
+}
+
+export interface ConfirmProps extends DialogProps {
+    /** Properties of the cancel button */
+    cancelButton: ButtonProps;
+}
+
+const { block, elem } = bem('Alert', styles);
+
+const Dialog: React.FC<AlertProps | ConfirmProps> = (props) => {
+    const { isOpen, children, title, acceptButton, cancelButton, ...rest } = props;
+
+    const isConfirm = !!cancelButton;
+
+    return (
+        <Modal {...rest} {...block(props)} isOpen={isOpen}>
+            <div {...elem('content', props)} role="alert">
+                {!!title && (
+                    <Heading level="h2" {...elem('title', props)}>
+                        {title}
+                    </Heading>
+                )}
+                {children}
+            </div>
+            <div {...elem('actions', props)}>
+                {isConfirm && (
+                    <Button onClick={cancelButton!.onClick} context="link">
```

Comment:
Time to upgrade typescript (not as part of this PR of course) :) 
TS 4.4 is smart enough to understand that `cancelButton` here is of `ConfirmProps` type, so no need for `!`.

But having TS 3.8 I'd rather use `cancelButton` instead of `isConfirm` here:
```
{cancelButton && (
                    <Button onClick={cancelButton.onClick} context="link">
```

Reply:
I was thinking to do that too, but I thought it is more readable this way. Even if the TS isn't that nice, the dev intention is easier to understand.

Reply:
I can add a TODO point to remove `!` when upgrading TS